File: book_review_webapi/model/book.go
package model

import "time"

type Book struct {
	Isbn        string     `json:"isbn" gorm:"primary_key;size:11;unique"`
	Title       string     `json:"title"`
	Description string     `json:"description"`
	ImageURL    string     `json:"image_url"`
	Author      string     `json:"author"`
	ReleaseDate time.Time  `json:"release_date"`
	Rating      float32    `json:"rating"`
	Reviews     []Review   `json:"reviews" gorm:"foreignKey:book_isbn"`
	Categories  []Category `json:"categories" gorm:"many2many:book_categories;"`
}

File: book_review_webapp/src/model/book.ts
import { CategoryResponse } from "./category"

export type BookResponse = {
    isbn: string
    title: string
    description: string
    image_url: string
    author: string
    release_date: Date
    categories: CategoryResponse[]
}

export type BookRequest = {
    isbn: string
    title: string
    description: string
    image: File | null
    author: string
    release_date: string
    categories: CategoryResponse[]
}

export type BookUpdate = {
    title: string
    description: string
    categories: CategoryResponse[]
}